### PR TITLE
Edit Makefile to support compiling code in subdirectories

### PR DESCRIPTION
**Pull request by Octogonapus**
Added Pixel's makefile to support subdirectories.

### CODE REVIEW

**Comment by HotelCalifornia** [comment]
Is this confirmed to be backwards-compatible? I.e. have you confirmed that this will properly compile code without subdirectories?

**Comment by Octogonapus** [comment]
Yes, I have tested a simple project with no subdirectories on the same robot I tested the more complex project on, and confirmed it is backwards-compatible.

**Comment by edjubuh** [comment]
What happens when I have something like `libblrs/main.c` and `libwpi/main.c`? How do object files get resolved? Pretty sure at a glance it uses the relative path with directory separators as underscores but just want to make sure (so `libblrs_main.o` and `libwpi_main.o`)

**Comment by Octogonapus** [comment]
All object files get generated with their full path as their name, so you can have files in multiple directories with the same name.

**Comment by edjubuh** [comment]
What was your more complicated project structure like?

**Comment by Octogonapus** [comment]
https://github.com/OkapiLib/OkapiLib

**Comment by edjubuh** [comment]
Nice library! Could you theoretically remove the Makefile inside src? I'm leading to asking if the wildcard `**/*.c` matches `src/libwpi/hello/main.c`, which I can't remember from Makefile documentation.

**Comment by Octogonapus** [comment]
You might be able to, but I'm not skilled enough in make to provide a definitive answer. I've poked this for a few days and haven't come up with anything.

File: template/src/Makefile [hunk 1]
```diff
@@ -39,12 +39,14 @@ $(ASMOBJ): $(BINDIR)/%.o: %.$(ASMEXT)
...
 
 # Object management
-$(COBJ): $(BINDIR)/%.o: %.$(CEXT) $(HEADERS) $(HEADERS_2)
+$(COBJ): $(BINDIR)/$(subst ..,,$(subst /,_,%.o)): %.$(CEXT) $(HEADERS) $(HEADERS_2)
 	@echo CC $(INCLUDE) $<
-	@$(CC) $(INCLUDE) $(CFLAGS) -o $@ $<
+	@$(CC) $(INCLUDE) $(CFLAGS) -o $(BINDIR)/$(subst ..,,$(subst /,_,$@)) $<
 
-$(CPPOBJ): $(BINDIR)/%.o: %.$(CPPEXT) $(HEADERS) $(HEADERS_2)
+KEKSRC:=$(dir $(wildcard **/*/))
```
Comment (edjubuh) [inline template/src/Makefile]: Is this being used?

**Comment by NoMod-Programming** [comment]
Not working with c++ files in subdirectories of src.

**Pull request closed without merging**